src/App.js

```diff
@@ -1,25 +1,49 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+
+import "./App.css";
+
+const App = () => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [changeBackgroundColor, setChangeBackgroundColor] = useState(true);
 
-function App() {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
+    <div
+      style={{ backgroundColor: changeBackgroundColor ? "white" : "darkgray" }}
+    >
+      <div className="App">
+        <input
+          type="text"
+          placeholder="Enter username"
+          value={username}
+          onChange={(event) => {
+            setUsername(event.target.value);
+          }}
+        />
+        <input
+          type={showPassword ? "text" : "password"}
+          placeholder="Enter password"
+          value={password}
+          onChange={(event) => {
+            setPassword(event.target.value);
+          }}
+        />
+
+        <button type="button" onClick={() => setShowPassword(!showPassword)}>
+          Show/Hide password
+        </button>
+        <button
+          type="button"
+          onClick={() => setChangeBackgroundColor(!changeBackgroundColor)}
         >
-          Learn React
-        </a>
-      </header>
+          Change background color
+        </button>
+        <p>{username}</p>
+        <p>{showPassword ? password : "*".repeat(password.length)}</p>
```

Comment:
This should be a util function - util functions are common functions that may be utilized in the whole app, 
    e.g.  "*".repeat(password.length)

extract  this functionality - "*".repeat(password.length) - to util function
1) In src, create a directory called utils
2) Create a file named stringUtils.js
3) Create a function with this functionality - "*".repeat(password.length)